### PR TITLE
threadpool: wait for threads to start

### PR DESCRIPTION
**Pull request by bzoz**
After some investigation of failures described in https://github.com/libuv/libuv/pull/1613#issuecomment-344958863, it looks like libuv can die if the app exits before worker threads from threadpool finish initialization. I believe this is the reason for fs and threadpool tests flakiness.

This makes `init_threads` wait for all threads to start before continuing. It fixes some test flakiness on Windows.

### CODE REVIEW

**Comment by bzoz** [comment]
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/583/

**Comment by bnoordhuis** [comment]
Fixes the Windows build but causes fallout across Unix machines.  I'll have a look.

File: test/test-loop-close.c [hunk 1]
```diff
@@ -55,3 +55,21 @@ TEST_IMPL(loop_close) {
...
+TEST_IMPL(loop_instant_close) {
+  uv_loop_t loop;
+  uv_work_t req;
+  ASSERT(0 == uv_loop_init(&loop));
+  ASSERT(0 == uv_queue_work(&loop,
+                            &req,
+                            loop_instant_close_work_cb,
+                            loop_instant_close_after_work_cb));
+  MAKE_VALGRIND_HAPPY();
+  return 0;
```
Comment (bnoordhuis) [inline test/test-loop-close.c]: This test as is cannot work: `loop` and `req` go out of scope (and get clobbered) but they're still in use.

**Comment by bzoz** [comment]
Added `static`, new CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/584/

**Comment by santigimeno** [comment]
FWIW, moving the lock from [here](https://github.com/janeasystems/libuv/blob/47c90c99d24f2b302c800d91e535cb6eef255768/src/threadpool.c#L180) up some lines just before the loop that creates the threads makes the tests pass.
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/586/

**Comment by bzoz** [comment]
Closing since https://github.com/libuv/libuv/pull/1639 landed

**Comment by bzoz** [comment]
Can't reopen after rebasing, new PR: https://github.com/libuv/libuv/pull/1660

**Pull request closed without merging**